projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -76,11 +149,73 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   onExpandWidth(e) {}
 
   onClickAccept(e) {
-    this.onAccept.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.acceptLoading = true;
+    this.groupService
+      .acceptJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "accept", "accepted");
+          this.acceptLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "accept");
+          this.acceptLoading = false;
+        }
+      );
   }
 
   onClickReject(e) {
-    this.onReject.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.rejectLoading = true;
+    this.groupService
+      .rejectJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "reject", "declined");
+          this.rejectLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "reject");
+          this.rejectLoading = false;
+        }
+      );
+  }
+
+  onSelectAll(event) {
+    if (this.selection.length === this.requests.length) {
+      this.selection = [];
+    } else {
+      this.selection = this.requests;
+    }
+  }
+
+  onHeaderCheckboxToggle(event) {
+    if (event.checked) {
+      this.selection = this.requests;
+    } else {
+      this.selection = [];
+    }
   }
 
   onCustomSort(event: SortEvent) {
```

Comment:
```
if .... { diff = true }
if (!diff) return;
something
```
What is it for a programming style?
The same as 
```
if ... { something }
```
Isn't it?

And normally, with RxJS, you write that without a superfluous state variable.

Reply:
ignored?

Reply:
Not done yet.
What should I do?

Reply:
fixed.